Automattic/jetpack
# My Jetpack: Add free search product support

Fixes #26767 

#### Changes proposed in this Pull Request:
- Add a status `needs_purchase_or_free` for product and set search to use it
- Add `wpcom_free_product_slug` for object `pricing_for_ui`
- Add `Start for free` button if `wpcom_free_product_slug` is not empty
- Adds `new_pricing_202208=1` in GET param to gate the feature

#### Other information:


#### Jetpack product discussion
pcNPJE-1f2-p2

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
* Apply this change to your Jetpack site.
* With a supported plugin installed (i.e. Jetpack or any of the standalone plugins), navigate to the My Jetpack page.
* Ensure that the label for Search has been updated to "Start for free" when `new_pricing_202208=1` added as a GET param.
* Ensure that the Search purchase page now features a new "Start for free" button below the "Add Jetpack Search" button when `new_pricing_202208=1` added as a GET param.

<img width="896" alt="image" src="https://user-images.githubusercontent.com/1425433/195512052-b5e16b76-bc72-4c4d-ba43-09da2ad6b7e6.png">
